components/style/lib.rs

```diff
@@ -72,6 +74,11 @@ extern crate time;
 extern crate url;
 extern crate util;
 
+#[cfg(feature = "gecko")]
+#[path = "./gecko_string_cache/mod.rs"]
+#[allow(unsafe_code)]
+#[macro_use] pub mod string_cache;
```

Comment:
Yeah, I understand why it's needed for gecko, but not for Servo, that's my point. `glue.rs` isn't compiled for servo.